src/ex2/PokemonClient.js

```diff
@@ -0,0 +1,179 @@
+        
+        if(this.itemManager.todoListSize() > 0){
+            this.updateUIWithNonEmptyInput()
+        }
+        else{
+            this.updateUIWithEmptyInput()
+        }
+    }
+
+    updateUIWithNonEmptyInput() {
+        const enterTodos = document.getElementById("enter-todos")
+        const clearAllTodosButton = document.getElementById("clear-all-todos-button")
+
+        clearAllTodosButton.classList.add("active")
+        clearAllTodosButton.style.cursor = "pointer"
+        enterTodos.style.display = "none"
+    }
+
+    updateUIWithEmptyInput() {
+        const enterTodos = document.getElementById("enter-todos")
+        const clearAllTodosButton = document.getElementById("clear-all-todos-button")
+
+        clearAllTodosButton.classList.remove("active")
+        clearAllTodosButton.style.cursor = "not-allowed"
+        enterTodos.style.display = "block"
+    }
+
+    createTodoListItems() {
+        const todoInput = document.getElementById("todo-input")
+
+        this.createItemsByCurrentData()
+        this.createItemsDeleteFuctionality()
+        
+        todoInput.value = "" //clear input
+    }
+
+    createItemsByCurrentData(){
+        const todoListElement = document.getElementById("todo-list")
+        let listItems = ""
+
+        this.itemManager.getTodoList().forEach((todo) => { 
+            listItems += `<li class="todo-item">${todo}
+                <div>
+                    <span class="action-btn delete">
+                        <i class="fas fa-trash"></i>
+                    </span>
+                </div>
+                </li>
+            `
+        })
+    
+        todoListElement.innerHTML = listItems
+    }
+
+    createItemsDeleteFuctionality(){
+        const deleteItems = document.querySelectorAll(".delete")
+
+        for (let i = 0; i < deleteItems.length; i++) {
+            deleteItems[i].addEventListener("click", () => this.deleteTodo(i))
+        }
+    }
+
+    deleteTodo(index){
+        const removedTodo = this.itemManager.deleteTodo(index)
+        alert(`removed new todo ${removedTodo}`) 
+    }
+
+    addTodo(){
+        const todoInput = document.getElementById("todo-input")
+        const addTodoButton = document.getElementById("add-todo-button")
+
+        let enterValue = todoInput.value
+        const inputIsEmpty = enterValue.trim() === ""
+
+        if(inputIsEmpty){
+            alert("todo cannot be empty")
+        }
+        else{
+            alert(`added new todo ${enterValue}`)
+        }
+
+        this.itemManager.addTodo(enterValue)
+        addTodoButton.classList.remove("active")
+    }
+
+    clearAllTodos(){
+        this.itemManager.clearAllTodos()
+        alert("all todos cleared")
+    }
+
+    filterDataAToZ() {
+        this.itemManager.filterDataAToZ()
+    }
+
+    filterDataZToA() {
+        this.itemManager.filterDataZToA()
+    }
+}
+
+document.addEventListener("DOMContentLoaded", function() {
```

Comment:
there shouldn't be any dom manipulation from here

Reply:
hi @shaneer  i fixed all your comments